src/dirs.proj

```diff
@@ -19,11 +19,44 @@
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
   </ItemGroup>
 
+  <Import Project="dir.targets" />
   <Import Project="..\dir.traversal.targets" />
 
   <PropertyGroup>
+    <SkipVersionGeneration>true</SkipVersionGeneration>
+    <!-- Turn off the automated package generation in packages.targets:AddNuGetPackageVersionMetadataToNuspecs;
+         we'll use the one produced by NerdBank.GitVersioning instead. -->
+    <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
+  </PropertyGroup>
+
+  <Import Project="$(ToolsDir)packages.targets" Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'" />
+
+  <PropertyGroup Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'">
+    <TraversalBuildDependsOn>
+      $(TraversalBuildDependsOn);
+      GetNuspecProperties;
+      BuildPackages;
+    </TraversalBuildDependsOn>
+    <!-- We use Microsoft.DotNet.Build tools to create our packages, but we want the version comes from 
+         NerdBack.GitVersioning. -->
+    <GetNuGetPackageVersionsDependsOn>SetPackageVersion</GetNuGetPackageVersionsDependsOn>
```

Comment:
Actually, looking closer, this seems a bit funky. `SetPackageVersion` below has `DependsOnTargets="GetNuGetPackageVersion"`, but by the usual convention, this would set the opposite `DependsOnTargets` on the `GetNuGetPackageVersions` target.

What is the dependency situation? My understanding from looking at this change is that you need to define a couple of properties after GitVersioning runs and before `BuildPackages` runs--is that correct? If so, you should be able to do that a bit more simply with something like

```XML
<Target Name="SetPackageVersion
        DependsOnTargets="GetNuGetPackageVersion"
        BeforeTargets="BuildPackages">
  <PropertyGroup>
    <PackageVersion>$(NuGetPackageVersion)</PackageVersion>
  </PropertyGroup>
  <ItemGroup>
    <!-- Used by packages.targets:BuildPackages to inject properties into the nuspec-->
    <NuspecProperties Include="version=$(NuGetPackageVersion)" />
  </ItemGroup>
</Target>
```

You'll still have to wire `BuildPackages` into the traversal build somehow (via `TraversalBuildDependsOn` as you've done here is fine).